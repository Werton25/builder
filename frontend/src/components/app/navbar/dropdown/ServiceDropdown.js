import React from 'react';
import PropTypes from 'prop-types';
import MethodDropdown from "./MethodDropdown";
import {fetchBuilder} from "../../../../actions/requests";
import {connect} from 'react-redux';
import {Dropdown} from 'bootstrap.native';

class ServiceDropdown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {services: []};
  }

  componentWillMount() {
    this.props.fetchBuilder().then((services) => this.setState({services}));
  }

  componentDidMount() {
    const dropdown = document.getElementById('service');
    !!dropdown && new Dropdown(dropdown, {persist: true});
  }

  render() {
    const methodDropdowns = this.state.services.map((service, i) => <MethodDropdown key={i} service={service}/>);
    return (
      <li className='dropdown'>
        <a className='dropdown-toggle' id='service' role='button'>
          Сервисы
          <span className='caret'/>
        </a>
        <ul className='dropdown-menu' role='menu'>
          {methodDropdowns}
        </ul>
      </li>
    );
  }
}

ServiceDropdown.propTypes = {
  fetchBuilder: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  fetchBuilder
};

export default connect(null, mapDispatchToProps)(ServiceDropdown);
