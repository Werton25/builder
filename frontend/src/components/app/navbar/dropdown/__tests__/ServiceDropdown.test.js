import React from "react";
import {BaseServiceDropdown} from "../ServiceDropdown";
import renderer from 'react-test-renderer';

describe('ServiceDropdown tests', () => {
  it('ServiceDropdown renders correctly', () => {
    const tree = renderer
      .create(<BaseServiceDropdown fetchBuilder={jest.fn().mockReturnValue(Promise.resolve([]))}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
