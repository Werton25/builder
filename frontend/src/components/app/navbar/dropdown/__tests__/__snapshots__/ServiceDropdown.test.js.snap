// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDropdown tests ServiceDropdown renders correctly 1`] = `
<li
  className="dropdown"
>
  <a
    className="dropdown-toggle"
    id="service"
    role="button"
  >
    Сервисы
    <span
      className="caret"
    />
  </a>
  <ul
    className="dropdown-menu"
    role="menu"
  />
</li>
`;
